### PR TITLE
Add-convert-to-var

### PR DESCRIPTION
**Pull request by alonkukl**
Adding the convert to variable untry pass
Adding test to that pass and also to the toU3 pass

### CODE REVIEW

File: bqskit/passes/util/converttou3.py [hunk 1]
```diff
@@ -17,17 +16,18 @@
 class ToU3Pass(BasePass):
     """Converts single-qubit general unitary gates to U3 Gates."""
 
+    def __init__(self, convert_all_single_qubit_gates: bool = False) -> None:
+        self.convert_all_single_qubit_gates = convert_all_single_qubit_gates
```
Comment (edyounis) [inline bqskit/passes/util/converttou3.py:20]: Need documentation here

File: bqskit/passes/util/converttovar.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+"""This module implements the ToU3Pass."""
```
Comment (edyounis) [inline bqskit/passes/util/converttovar.py:1]: Incorrect module docstring

File: bqskit/passes/util/converttovar.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+from bqskit.ir.point import CircuitPoint
+
+_logger = logging.getLogger(__name__)
+
+
+class ToVariablePass(BasePass):
+    """Converts single-qudit general unitary gates to Variable Unitary Gates."""
+
+    def __init__(self, convert_all_single_qudit_gates: bool = False) -> None:
+        self.convert_all_single_qudit_gates = convert_all_single_qudit_gates
```
Comment (edyounis) [inline bqskit/passes/util/converttovar.py:20]: Need documentation

File: tests/passes/util/test_to_u3.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    circuit = Circuit(1)
+    gate_utry = UnitaryMatrix(gen_random_utry_np(2))
+    circuit.append_gate(
+        single_qubit_general_gate, (0,),
+        single_qubit_general_gate.calc_params(gate_utry),
+    )
+
+    circuit = compiler.compile(circuit, [ToU3Pass()])
+    dist = circuit.get_unitary().get_distance_from(gate_utry)
+    assert dist <= 1e-7  # Is this good enough??
```
Comment (edyounis) [inline tests/passes/util/test_to_u3.py:34]: Yes

**Review by edyounis — CHANGES_REQUESTED** [review]

File: bqskit/passes/util/converttou3.py (hunk 1 above)
Comment (alonkukl) [reply bqskit/passes/util/converttou3.py:28]: Added and pushed

File: bqskit/passes/util/converttovar.py (hunk 2 above)
Comment (alonkukl) [reply bqskit/passes/util/converttovar.py]: Fixed it

File: bqskit/passes/util/converttovar.py (hunk 3 above)
Comment (alonkukl) [reply bqskit/passes/util/converttovar.py:27]: Added and pushed

**Review by edyounis — APPROVED** [review]

**Pull request merged**